tools/ci/check_c_api_usage.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/env python3
+from __future__ import annotations
```

Comment:
Is this just needed in case someone is using an older Python version than we normally support outside of linting?

Reply:
Yeah, I think intiially the issue popped up because I didn't think about a certain environment where this script may be used. So I thought maybe this would be helpful.

Reply:
I am open to removing this. I don't know if my initial assumption is correct or appropriate for the codebase. But lmk what your preference is here.